### PR TITLE
Added low and medium XXE to DVWA

### PR DESCRIPTION
**Pull request by MaartenPentinga**
Added low and medium XXE to DVWA.

Low -> basic XXE
Medium -> removed the output, so the attacker is forced to OOB XXE.

### CODE REVIEW

File: vulnerabilities/xxe/index.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+$vulnerabilityFile = '';
+switch( $_COOKIE[ 'security' ] ) {
+	case 'low':
+		$vulnerabilityFile = 'low.php';
+		break;
+	case 'medium':
+		$vulnerabilityFile = 'medium.php';
+		break;
+	case 'high':
+		$vulnerabilityFile = 'medium.php';
```
Comment (ethicalhack3r) [inline vulnerabilities/xxe/index.php]: Shouldn't this be `high.php`?

**Comment by digininja** [comment]
Does this really work? I thought the lower level PHP libraries killed off XXE in this way years ago. Would be good if it does work.

The help file needs examples of how to exploit the issue as well.

You need to remove your config files from the commit and come up with hard and impossible levels.

Could be a good start.

File: vulnerabilities/xxe/index.php (hunk 1 above)
Comment (digininja) [reply vulnerabilities/xxe/index.php]: He hasn't written high or impossible levels.

**Comment by MaartenPentinga** [comment]
Heya, 

Alright. I can add the exploit to the help file. To setup 'hard' and 'impossible' i will need a bit. 

Will remove the config files on the next pull. 

About the PHP version, running 7.2. Is there a certain version you want me to use? I checked the README.md, but it did't specify. 

Kind regards,
Maarten

**Comment by digininja** [comment]
I'd say go for whatever PHP version Kali uses as that is what most people seem to deploy DVWA into.

File: vulnerabilities/xxe/index.php (hunk 1 above)
Comment (MaartenPentinga) [reply vulnerabilities/xxe/index.php]: I left it to be at medium. This way regular XXE and OOB XXE can be practised.

**Comment by MaartenPentinga** [comment]
@digininja will do! 

Will get back at it later. 

I also added a seperate CSRF with POST request rather then GET. Only low difficulty till now.

**Comment by digininja** [comment]
Do that as a separate pull request otherwise things will get very mixed up.

**Comment by MaartenPentinga** [comment]
Will do, thanks for the tip (:

**Comment by digininja** [comment]
Why are you changing the xss_r help file?

**Comment by MaartenPentinga** [comment]
Made a small error and fixed that. Can send a new commit to remove it if you like, or make a new pull.

**Comment by digininja** [comment]
Go for a new PR as you need to remove the CSRF stuff as well.

**Comment by digininja** [comment]
And you need to remove your config files and not remove our htaccess and
gitignore.

**Comment by MaartenPentinga** [comment]
Strange, did remove the config files and readded the .htaccess. Will check what is going wrong there and then make a new pull request.

**Comment by MaartenPentinga** [comment]
Made a new pull, can you guys close this one? Cheers!

**Pull request closed without merging**